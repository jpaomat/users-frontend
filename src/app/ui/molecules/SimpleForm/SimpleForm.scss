.simple-form {
    display: block;
    margin: 4px;
}

.form-control:focus {
    box-shadow: 0 0 0 0.1rem rgba(13,110,253,.25);
}