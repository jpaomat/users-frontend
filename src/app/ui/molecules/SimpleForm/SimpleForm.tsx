import { ChangeEvent } from 'react';
import { Button, InputGroup, Form } from 'react-bootstrap';
import { FormDropdownProps } from '../../../interfaces';
import { useForm } from '../../../hooks/general/useForm';
import './SimpleForm.scss';

export const SimpleForm = ({ name, placeholder, buttonText, getFormFields, onCloseForm }: FormDropdownProps) => {
	const { formState, onInputChange, onReset } = useForm({
		[name]: ''
	});
	const onClick = () => {
		getFormFields(formState);
		onCloseForm && onCloseForm();
        onReset();
	};

	const onChange = (e: ChangeEvent<HTMLInputElement>) => {
		onInputChange(e);
	};

	return (
			<div className='simple-form'>
				<InputGroup size='sm'>
					<Form.Control name={name} value={formState[name]} placeholder={placeholder} onChange={onChange} />
					<Button variant='outline-secondary' id='button-addon2' onClick={onClick}>
						{buttonText}
					</Button>
				</InputGroup>
			</div>
	);
};
