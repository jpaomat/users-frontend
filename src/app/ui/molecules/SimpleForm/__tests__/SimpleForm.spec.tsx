import { fireEvent, render } from '@testing-library/react';
import { SimpleForm } from '../SimpleForm';
import { defaultMock } from './mocks';

describe('<SimpleForm /> component', () => { 
    it('should render component', () => {
		const {container} = render(<SimpleForm {...defaultMock} />);
        expect(container.querySelector('input')?.value).toBe(defaultMock.inputValue);
		expect(container).toMatchSnapshot();
	});

	it('should cleared the input when the clear button is clicked', () => {
		const {container} = render(<SimpleForm {...defaultMock} />);
		const buttonElement = container.querySelector('button');
        fireEvent.click(buttonElement as Element);
        expect(container.querySelector('input')?.value).toBe(defaultMock.inputValue);
	});
 });
