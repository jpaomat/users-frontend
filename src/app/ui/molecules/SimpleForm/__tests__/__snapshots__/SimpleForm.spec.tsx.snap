// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleForm /> component should render component 1`] = `
<div>
  <div
    class="simple-form"
  >
    <div
      class="input-group input-group-sm"
    >
      <input
        class="form-control"
        name="inputText"
        value="hi test"
      />
      <button
        class="btn btn-outline-secondary"
        id="button-addon2"
        type="button"
      >
        send
      </button>
    </div>
  </div>
</div>
`;
